projects/igniteui-angular/src/lib/grids/common/pipes.ts

```diff
@@ -355,3 +355,39 @@ export class IgxColumnFormatterPipe implements PipeTransform {
         return formatter(value);
     }
 }
+
+@Pipe({
+    name: 'gridAddRow',
+    pure: true
+})
+export class IgxGridAddRowPipe implements PipeTransform {
+
+    constructor(private gridAPI: GridBaseAPIService<IgxGridBaseDirective & GridType>) { }
+
+    transform(collection: any, parentID: number, enableAddRow: boolean, cancelAddMode: boolean, pipeTrigger: number) {
+        const grid = this.gridAPI.grid;
+        if (!enableAddRow) {
+            return collection;
+        }
+        if (parentID === -1) {
+            const result = collection.map(record => {
+                return grid.isAddRowRecord(record) ? record.recordRef : record;
+            });
+            grid.data = result;
+            return result;
+        }
+        const parentRecordIndex = collection.findIndex(record => record[grid.primaryKey] === parentID);
+        if (cancelAddMode) {
```

Comment:
Really unsure about this. Is the add row (while being in edit mode) ever part of the collection before this pipe is triggered? Don't you want to add this row _only_ when the state is "add row UI present". Once the user presses esc, you say "add row UI no longer present" and escape the pipe never reaching a point that adds the row.